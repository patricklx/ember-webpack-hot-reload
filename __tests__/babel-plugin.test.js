const plugin = require('../dist/lib/babel-plugin').default;
const babel = require('@babel/core');


describe('convert template with hot reload helpers', () => {
  it('should convert correctly', () => {
    const template = `
  import precompileTemplate from '@ember/template-compilation';
  precompileTemplate(\`
  {{page-title "HotReload"}}

{{! The following component displays Ember's default welcome message. }}
<br />
{{#let (helper 'print-number' x=1) as |output|}}
  helpler {{output}}
{{/let}}
{{! Feel free to remove this! }}

{{component 'my-comp' rootEventType=(or @rootEventType 'click')}}


{{#let this.FirstComponent as |output|}}
  {{output}}
{{/let}}

<div {{this.myModifier}}></div>

{{FirstComponent}}

<br />
26

<br />

{{this.x}}


{{outlet}}
  \`);`
    const result = babel.transform(template, {
      filename: '.embroider/rewritten-app/file.hbs',
      plugins: [plugin]
    });
    expect(result.code).toMatchSnapshot();
  })

  it('should convert basic-dropdown', () => {
    const tpl = `
    {{#let
  (hash
    uniqueId=this.publicAPI.uniqueId
    isOpen=this.publicAPI.isOpen
    disabled=this.publicAPI.disabled
    actions=this.publicAPI.actions
    Trigger=(if
      (eq @triggerComponent undefined)
      (component
        'basic-dropdown-trigger'
        dropdown=(readonly this.publicAPI)
        hPosition=(readonly this.hPosition)
        renderInPlace=(readonly this.renderInPlace)
        vPosition=(readonly this.vPosition)
      )
      (component
        (ensure-safe-component @triggerComponent)
        dropdown=(readonly this.publicAPI)
        hPosition=(readonly this.hPosition)
        renderInPlace=(readonly this.renderInPlace)
        vPosition=(readonly this.vPosition)
      )
    )
    Content=(if
      ((helper 'eq') @contentComponent undefined)
      (component
        'basic-dropdown-content'
        dropdown=(readonly this.publicAPI)
        hPosition=(readonly this.hPosition)
        renderInPlace=(readonly this.renderInPlace)
        preventScroll=(readonly @preventScroll)
        rootEventType=(or @rootEventType 'click')
        vPosition=(readonly this.vPosition)
        destination=(readonly this.destination)
        top=(readonly this.top)
        left=(readonly this.left)
        right=(readonly this.right)
        width=(readonly this.width)
        height=(readonly this.height)
        otherStyles=(readonly this.otherStyles)
      )
      (component
        (ensure-safe-component @contentComponent)
        dropdown=(readonly this.publicAPI)
        hPosition=(readonly this.hPosition)
        renderInPlace=(readonly this.renderInPlace)
        preventScroll=(readonly @preventScroll)
        rootEventType=(or @rootEventType 'click')
        vPosition=(readonly this.vPosition)
        destination=(readonly this.destination)
        top=(readonly this.top)
        left=(readonly this.left)
        right=(readonly this.right)
        width=(readonly this.width)
        height=(readonly this.height)
        otherStyles=(readonly this.otherStyles)
      )
    )
  )
as |api|
}}
  {{#if this.renderInPlace}}
    <div class='ember-basic-dropdown' ...attributes>{{yield api}}</div>
  {{else}}
    {{yield api}}
  {{/if}}
{{/let}}`;
    const code = `
    import precompileTemplate from '@ember/template-compilation';
    precompileTemplate(\`${tpl}\`);
    `;
    const result = babel.transform(code, {
      filename: '.embroider/rewritten-app/file.hbs',
      plugins: [plugin]
    });
    expect(result.code).toMatchSnapshot();
  })
});