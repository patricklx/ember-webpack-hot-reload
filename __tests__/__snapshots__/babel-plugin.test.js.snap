// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert template with hot reload helpers should convert correctly 1`] = `
"import precompileTemplate from '@ember/template-compilation';
precompileTemplate(\`
  {{#let (webpack-hot-reload (helper "page-title") type="helper") as |helper_0|}}{{helper_0 "HotReload"}}{{/let}}

{{!-- The following component displays Ember's default welcome message. --}}<br />
{{#let (webpack-hot-reload (helper "print-number") type="helper") as |output|}}  helpler {{output}}
{{/let}}{{!-- Feel free to remove this! --}}
{{#let (webpack-hot-reload (component "my-comp") type="component") as |component_2|}}{{component_2 rootEventType=((webpack-hot-reload (helper "or") type="helper") @rootEventType "click")}}{{/let}}


{{#let this.FirstComponent as |output|}}  {{output}}
{{/let}}
<div {{this.myModifier}}></div>

{{#let (webpack-hot-reload (component "FirstComponent") type="component") as |component_4|}}{{component_4}}{{/let}}

<br />
26

<br />

{{this.x}}


{{outlet}}
  \`);"
`;
